### PR TITLE
#161342847 Tag new articles

### PR DESCRIPTION
**Pull request by victor-nweze-deel**
#### What does this PR do?
Tag new articles

#### Description of Task to be completed?
- Create new models and migrations
- Modify the create article route to accomodate tags
- Update join table
- Write test

#### How should this be manually tested?
- On the terminal run git clone https://github.com/andela/haven-ah-backend.git.
- Run git checkout ft-tag-articles-161342847
- Run `npm install` to install the npm packages.
- Run `npm test` to run the unit and integration tests.

To test on postman
- run `npm run start-dev` and open up postman.
- Hit the route `localhost:5000/api/v1/articles`

#### What are the relevant pivotal tracker stories?
- [#161342847](https://www.pivotaltracker.com/story/show/161342847)

#### Screenshots (if appropriate)
screen shot 2018-11-02 at 4 21 19 pm
screen shot 2018-11-05 at 10 41 15 am
screen shot 2018-11-05 at 10 41 50 am

### CODE REVIEW

**Comment by PriscillaSam** [comment]
This looks good. Good work @ProfJigsaw.

**Review by TheoOkafor — COMMENTED** [review]
Nice work Victor

File: server/repository/tagRepository.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  /**
+ * Function to create a new tag
+ * @param {string} tagName
+ * @param {number} articleId
+ * @returns {object} tag
+ */
+  static async createTag(tagName, articleId) {
+    const tag = await Tags.findOrCreate({
+      where: {
+        tagName,
```
Comment (TheoOkafor) [inline server/repository/tagRepository.js:18]: Can we have a way of handling tag names that are not submitted as lowercase?
So we do not have duplicated tag names with different cases.

Comment (victor-nweze-deel) [reply server/repository/tagRepository.js:18]: Nice catch @TheoOkafor I will see to this.

Comment (victor-nweze-deel) [reply server/repository/tagRepository.js:18]: Fixed

File: server/controllers/article.js [hunk 2]
```diff
@@ -27,9 +29,24 @@ class Article {
         userid: request.userId,
         description,
         readtime,
-        images: images.split(','),
+        images: images ? images.split(',') : [],
       });
-      return goodHttpResponse(response, 201, 'Article Created', newArticle);
+      if (request.body.tags) {
+        const combinedTags = request.body.tags.toLowerCase();
```
Comment (seyi-adeleke) [inline server/controllers/article.js:35]: is tags a string?

**Comment by seyi-adeleke** [comment]
please document this endpoint

File: server/controllers/article.js (hunk 2 above)
Comment (victor-nweze-deel) [reply server/controllers/article.js:35]: Yes, it is. They are bundled and sent as a string. I then split the tags into an array

**Comment by victor-nweze-deel** [comment]
> please document this endpoint

I have updated the swagger.js file. Creating tags works with the create article routes and so it doesn't have a different route handler.

**Pull request merged**